creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -123,31 +153,59 @@ const updateReplicaSetJobProcessor = async function ({
       replicaToUserInfoMap,
       enabledReconfigModes
     })
-    ;({ errorMsg, issuedReconfig, syncJobsToEnqueue } =
-      await _issueUpdateReplicaSetOp(
-        userId,
-        wallet,
-        primary,
-        secondary1,
-        secondary2,
-        newReplicaSet,
-        audiusLibs,
-        logger
-      ))
-  } catch (e: any) {
+
+    try {
+      ;({ errorMsg, issuedReconfig, syncJobsToEnqueue, error } =
+        await _issueUpdateReplicaSetOp(
+          userId,
+          wallet,
+          primary,
+          secondary1,
+          secondary2,
+          newReplicaSet,
+          audiusLibs,
+          logger
+        ))
+    } catch (e) {
+      error = error || 'failure_issue_update_replica_set'
+      logger.error(
+        `ERROR issuing update replica set op: userId=${userId} wallet=${wallet} old replica set=[${primary},${secondary1},${secondary2}] | Error: ${(
+          e as Error
+        ).toString()}: ${(e as Error).stack}`
+      )
+      errorMsg = (e as Error).toString()
+    }
+  } catch (e) {
+    error = error || 'failure_determine_new_replica_set'
```

Comment:
this pattern of `error = error || <error_string>` seems very error prone
if error is already set in some earlier try catch, we'd just never overwrite it?
the safer approach i think would be do what `issueSyncRequest.jobProcessor` does with an internal `handle` function that has a return statement in each `catch` block, which is parsed by the caller to set and return value

Reply:
I wanted to do the same but seemed to be the possibility for multiple levels or errors to happen. 

Doing `error = error || <error_string>` seemed like the quickest way to add the metrics without rewriting the job processor. 

I can extend the scope of the ticket to refactoring the job processor if you think that's the better approach though

Reply:
curious what scope of refactor you think would be required to get around this - i'm just basing off `issuesyncRequest` which i think is a somewhat straightfwd refactor. do you think that pattern would work here?